tools/fs/files.js

```diff
@@ -737,6 +745,32 @@ files.extractTarGz = function (buffer, destPath, options) {
   }
 };
 
+function waitFSToSync(dir) {
+	let waitToSync = true;
+
+	while (waitToSync) {
```

Comment:
This should have a sleep cycle and a maximum spin time—it cannot run unbounded like this, possibly forever if the file system fails to cooperate.  As written right now this could run an enormous number of file operations!